### ISSUE TITLE
Series data points are not reachable on mouse hover in line and area charts with Firefox

### ISSUE THREAD

**Issue by tobigit** [swimlane/ngx-charts#469]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
I try to move the mouse cursor over a data point on a line- or area-chart with firefox. But the series-group-tooltip always overlap the data-point I want to reach.

**Expected behavior**
All data points should be reachable with the mouse cursor.

**Reproduction of the problem**
Move the mouse cursor to any data point with firefox in this plunker:
https://plnkr.co/edit/WTqX5jmYzCmjZI16gkg9?p=preview

**Please tell us about your environment:**
Windows 7, IntelliJ

* **ngx-charts version:** 6.0.0

* **Angular version:** 4.2.5

* **Browser:** Firefox 54.0.1

* **Language:** all

**Comment by AnnaGranovsky** [comment]
I have the same problem here on firefox with area chart

**Comment by CramericaIndustries** [comment]
+1
any workarounds?

**Comment by weequ** [comment]
I think I have the same issue. When I move mouse over the tooltip area it always shows tooltip of an element that has lower x coordinate. Works perfectly in chromium browser.

I think the issue might be in https://github.com/swimlane/ngx-charts/blob/master/src/common/tooltip-area.component.ts#L148.

According to https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/offsetX the offsetX property is experimental and should not be used in production.

**Comment by CramericaIndustries** [comment]
Error is still present in Firefox 57.0 (Quantum).

**Comment by marjan-georgiev** [comment]
Does upgrading to ngx-charts 7.0.1 fix this?

**Comment by CramericaIndustries** [comment]
Do I need to upgrade to ng5 to use ngx-charts 7.0.1? Or is it compatible to ng4.4.6?

**Comment by marjan-georgiev** [comment]
Yes, you need angular 5 for ngx-charts v7.

**Comment by cesco69** [comment]
Is there any workaround for this problem?

**Comment by Luukschoen** [comment]
As @cesco69 said, this issue still exists (despite v7.0.1 and new Firefox). @marjan-georgiev could you give us some pointers on how to solve this?

**Comment by tobigit** [comment]
The Problem still exists with latest version of ngx-charts (v7.0.1) and angular 5.

**Comment by cesco69** [comment]
Same problem exists for IE 11 also

**Comment by CramericaIndustries** [comment]
Same glitch in Edge Browser 16.16299 on Windows 10 Pro as well

**Comment by Luukschoen** [comment]
Okay so for anyone that got stuck here. I just examined the code and came to the conclusion that as @weequ mentioned offsetX in tooltip-area.component.ts could be the culprit here.

Examining the xPos on the mouseMove event, I came to notice that Firefox/IE gave me back entirely different values for event.offsetX. Which led to xPositions that were below zero. 

I'm talking about this rule exactly: 

 `const xPos = event.offsetX - this.dims.xOffset;`

Since I needed a really quick fix I build a small browser check around it to correct the wrong calculation, but that doesn't feel like the trick to solve this issue in general. Anyone has a better idea on how to deal with this mis positioning?

**Comment by Luukschoen** [comment]
So apparently in Chrome 

event.pageX - event.target.getBoundingClientRect().left gives me a different value then event.offsetX as where as in Firefox these two result in exactly the same numbers. The difference between the numbers in Chrome is compensated by subtracting this.dims.xOffset. To make this work in both browsers we could either check in which browser we are and decide whether to subtract or not to subtract dims.xOffset, or we could do a different calculation. Anyone has a suggestion for what's best?

**Comment by CramericaIndustries** [comment]
@Luukschoen did it work?

**Comment by Luukschoen** [comment]
@CramericaIndustries yes it did. I will create a PR in 1, 2, 3 ..

**Comment by CramericaIndustries** [comment]
@marjan-georgiev will @Luukschoens fix be included in the next release?

**Comment by CramericaIndustries** [comment]
If somebody wants to apply the patch manually until the new release is out:
Open the file `<project-dir>/node_modules/@swimlane/ngx-charts/release/index.js`
Then find and replace the line
`const xPos = event.offsetX - this.dims.xOffset;`
with
`const xPos = event.pageX - event.target.getBoundingClientRect().left;`
(that is in the `TooltipArea.prototype.mouseMove`  function)

**Comment by marjan-georgiev** [comment]
This was added and released in 7.1.0
